test/Microsoft.TemplateEngine.Core.UnitTests/ChunkStreamReadTests.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Core.Contracts;
+using Microsoft.TemplateEngine.Core.Operations;
+using Microsoft.TemplateEngine.Core.Util;
+using Microsoft.TemplateEngine.Mocks;
+using Microsoft.TemplateEngine.TestHelper;
+using Xunit;
+
+namespace Microsoft.TemplateEngine.Core.UnitTests
+{
+    public class ChunkStreamReadTests : TestBase, IClassFixture<EnvironmentSettingsHelper>
+    {
+        private IEngineEnvironmentSettings _engineEnvironmentSettings;
+
+        public ChunkStreamReadTests(EnvironmentSettingsHelper environmentSettingsHelper)
+        {
+            _engineEnvironmentSettings = environmentSettingsHelper.CreateEnvironment(hostIdentifier: this.GetType().Name, virtualize: true);
+        }
+
+        [Fact]
+        public void VerifyLongStreamNoReplacement()
+        {
+            Random rnd = new Random();
+            byte[] valueBytes = new byte[1024 * 1024];
+            rnd.NextBytes(valueBytes);
+            ChunkMemoryStream input = new ChunkMemoryStream(valueBytes, 1024);
+            ChunkMemoryStream output = new ChunkMemoryStream(1024);
```

Comment:
Do we have to use 1024 on both? Maybe randomizing these could catch some other bugs.
```suggestion
            ChunkMemoryStream input = new ChunkMemoryStream(valueBytes, 594);
            ChunkMemoryStream output = new ChunkMemoryStream(960);
```